src/Native/System.Security.Cryptography.Native/pal_x509.cpp

```diff
@@ -33,3 +73,182 @@ extern "C" X509* DecodeX509(const unsigned char* buf, int32_t len)
 
     return d2i_X509(nullptr, &buf, len);
 }
+
+extern "C" int32_t GetX509DerSize(X509* x)
+{
+    return i2d_X509(x, nullptr);
+}
+
+extern "C" int32_t EncodeX509(X509* x, uint8_t* buf)
+{
+    return i2d_X509(x, &buf);
+}
+
+extern "C" void X509Destroy(X509* a)
+{
+    if (a != nullptr)
+    {
+        X509_free(a);
+    }
+}
+
+extern "C" X509* X509Duplicate(X509* x509)
+{
+    return X509_dup(x509);
+}
+
+extern "C" X509* PemReadX509FromBio(BIO* bio)
+{
+    return PEM_read_bio_X509_AUX(bio, nullptr, nullptr, nullptr);
+}
+
+extern "C" ASN1_INTEGER* X509GetSerialNumber(X509* x509)
+{
+    return X509_get_serialNumber(x509);
+}
+
+extern "C" X509_NAME* X509GetIssuerName(X509* x509)
+{
+    return X509_get_issuer_name(x509);
+}
+
+extern "C" X509_NAME* X509GetSubjectName(X509* x509)
+{
+    return X509_get_subject_name(x509);
+}
+
+extern "C" int32_t X509CheckPurpose(X509* x, int32_t id, int32_t ca)
+{
+    return X509_check_purpose(x, id, ca);
+}
+
+extern "C" int32_t X509CheckIssued(X509* issuer, X509* subject)
+{
+    return X509_check_issued(issuer, subject);
+}
+
+extern "C" uint64_t X509IssuerNameHash(X509* x)
+{
+    return X509_issuer_name_hash(x);
+}
+
+extern "C" int32_t X509GetExtCount(X509* x)
+{
+    return X509_get_ext_count(x);
+}
+
+extern "C" X509_EXTENSION* X509GetExt(X509* x, int32_t loc)
+{
+    return X509_get_ext(x, loc);
+}
+
+extern "C" ASN1_OBJECT* X509ExtensionGetObject(X509_EXTENSION* x)
```

Comment:
In my parallel change for X500DistinguishedName I used the suffix "Oid" here instead of "Object" since what they're really returning is an object identifier (aka OID :)).